stl/inc/memory

```diff
@@ -1504,6 +1556,238 @@ private:
+        ++_Last;
+    }
+
+    _NoThrowIt _Release() { // suppress any exception handling backout and return _Last
+        _First = _Last;
+        return _Last;
+    }
+};
+
+template <class _Ty>
+void _Reverse_destroy_multidimensional_n(_Ty* const _Arr, const size_t _Size) {
+    if (_Size != 0) {
+        size_t _Idx = _Size;
+        do {
+            --_Idx;
+            // TRANSITION, LLVM-42404
+            if constexpr ((rank_v<_Ty>) > 0) {
+                _Reverse_destroy_multidimensional_n(_Arr[_Idx], extent_v<_Ty>);
+            } else {
+                _Destroy_in_place(_Arr[_Idx]);
+            }
+        } while (_Idx > 0);
+    }
+}
+
+template <class _Ty, size_t _Size>
+void _Uninitialized_copy_multidimensional(const _Ty (&_In)[_Size], _Ty (&_Out)[_Size]) {
+    // TRANSITION, LLVM-42404
+    if constexpr ((rank_v<_Ty>) > 0) {
+        for (size_t _Idx = 0; _Idx < _Size; ++_Idx) {
+            _TRY_BEGIN
+            _Uninitialized_copy_multidimensional(_In[_Idx], _Out[_Idx]);
+            _CATCH_ALL
+            _Reverse_destroy_multidimensional_n(_Out, _Idx);
+            _RERAISE;
+            _CATCH_END
+        }
+    } else {
+        if constexpr (_Ptr_copy_cat<_Ty*, _Ty*>::_Really_trivial) {
+            _Copy_memmove(_In, _In + _Size, _Out);
+        } else {
+            _Uninitialized_rev_destroying_backout<_Ty*> _Backout{_Out};
+            for (size_t _Idx = 0; _Idx < _Size; ++_Idx) {
+                _Backout._Emplace_back(_In[_Idx]);
+            }
+            _Backout._Release();
+        }
+    }
+}
+
+template <class _Ty>
+void _Uninitialized_value_construct_multidimensional_n(_Ty* const _Out, const size_t _Size) {
+    // TRANSITION, LLVM-42404
+    if constexpr ((rank_v<_Ty>) > 0) {
+        for (size_t _Idx = 0; _Idx < _Size; ++_Idx) {
+            _TRY_BEGIN
+            _Uninitialized_value_construct_multidimensional_n(_Out[_Idx], extent_v<_Ty>);
+            _CATCH_ALL
+            _Reverse_destroy_multidimensional_n(_Out, _Idx);
+            _RERAISE;
+            _CATCH_END
+        }
+    } else {
+        if constexpr (_Use_memset_value_construct_v<_Ty*>) {
+            _Zero_range(_Out, _Out + _Size);
+        } else {
+            _Uninitialized_rev_destroying_backout<_Ty*> _Backout{_Out};
+            for (size_t _Count = 0; _Count < _Size; ++_Count) {
+                _Backout._Emplace_back();
+            }
+            _Backout._Release();
+        }
+    }
+}
+
+template <class _Ty>
+void _Uninitialized_fill_multidimensional_n(_Ty* const _Out, const size_t _Size, const _Ty& _Val) {
+    // TRANSITION, LLVM-42404
+    if constexpr ((rank_v<_Ty>) > 0) {
+        for (size_t _Idx = 0; _Idx < _Size; ++_Idx) {
+            _TRY_BEGIN
+            _Uninitialized_copy_multidimensional(_Val, _Out[_Idx]);
+            _CATCH_ALL
+            _Reverse_destroy_multidimensional_n(_Out, _Idx);
+            _RERAISE;
+            _CATCH_END
+        }
+    } else {
+        if (0 < _Size) {

```

Comment:
That check feels more as a precondition. It is trivially fulfillled by the above loop but maybe make it an early return`?

Reply:
This check is now made only if function from `memset` family is used. I might be wrong but I suppose that these function need it because these checks are made in `uninitialized_***` algorithms.